server/service/device_client.go

```diff
@@ -91,3 +80,37 @@ func (dc *DeviceClient) Ping() error {
 
 	return err
 }
+
+func (dc *DeviceClient) listDevicePolicies(token string) ([]*fleet.HostPolicy, error) {
+	verb, path := "GET", "/api/latest/fleet/device/"+token+"/policies"
+	var responseBody listDevicePoliciesResponse
+	err := dc.request(verb, path, "", &responseBody)
+	return responseBody.Policies, err
+}
+
+func (dc *DeviceClient) NumberOfFailingPolicies(token string) (uint, error) {
+	verb, path := "GET", "/api/latest/fleet/device/"+token+"/desktop"
+	var r FleetDesktopResponse
+	err := dc.request(verb, path, "", &r)
+
+	if err == nil {
+		return uintValueOrZero(r.FailingPolicies), nil
+	}
+
+	if err != nil && dc.GetServerCapabilities().Has(fleet.CapabilityDesktopEndpoint) {
```

Comment:
Not too familiar with the Fleet Desktop implementation and the capabilities support, but I'm curious why we make the call to the minimal endpoint before checking if it has that capability? Is that to prevent the request to `GetServerCapabilities` in the more common case of the server being as up-to-date as FD?

Reply:
@mna The capabilities header is set on each req-resp cycle, by issuing a request to the minimal endpoint first we are guarantied that the resp header will have the latest capabilities - I hope that makes sense.

@lucasmrod Sounds good, the least code the better - I'll refactor this